app/models/user.rb

```diff
@@ -8,6 +8,17 @@ def set_default_role
     self.role ||= :user
   end
 
+  # Track the logged-in user at the model level
+  # so that we can leverage the user's organization
+  # within blazer reporting.
+  def self.current
+    Thread.current[:user]
```

Comment:
Curious whether a gem like [request_store](https://github.com/steveklabnik/request_store) could be a good idea here. I haven't used it though, just throwing the idea out there.

Reply:
Curious why we need to tap into `Thread` here. Is there something about just using `current_user` directly that wont work?

Reply:
This is a thread-safe way to keep track of the currently logged-in user at the model level, which doesn't have access to controller/view level variables like `current_user`. I just know from experience that in this context, if you don't store it on `Thread`, the value would be non-deterministic if you're running on a threaded web server.

Reply:
Ahh, got it. I see now that this is being used in an initializer. Thought on using Rails built in [ActiveSupport::CurrentAttributes](https://api.rubyonrails.org/classes/ActiveSupport/CurrentAttributes.html) for this?

Reply:
Our main app isn't up on Rails 6 yet, so the `CurrentAttributes` is something I wasn't familiar with. Seems like it could work. I can try it.